JSONResourceFile.js

```diff
@@ -189,26 +233,28 @@ JSONResourceFile.prototype.getContent = function() {
 /**
  * @private
  */
-JSONResourceFile.prototype._calcLocalePath = function(locale) {
+JSONResourceFile.prototype._calcLocalePath = function(locale, filename) {
     var rootLocale = "en-US";
-    var lo = new Locale(locale);
-    var fullPath = "";
-
+    var loc = new Locale(locale);
+    var lo = loc.getSpec();
+    var path = "";
+    var resDir = this.project.getResourceDirs("json")[0] || ".";
+    var mappingData = this.getMappings(filename);
+    
     if (this.baseLocale) {
-        if (this.locale.getSpec() !== rootLocale) {
-            fullPath = "/" + lo.getLanguage();
-        }
-    } else {
-        var nodeVersion = process.versions["node"].split(".")[0];
-        if (nodeVersion < 15) {
-            fullPath += "/" + locale.getSpec().replace(/-/g, "/");
-        } else {
-            // replaceAll() is available since v15.0.0
-            fullPath += "/" + locale.getSpec().replaceAll("-", "/");
+        if (locale !== rootLocale) {
+            lo = loc.getLanguage();
         }
-        
     }
-    return fullPath;
+    var path = this.API.utils.formatPath(mappingData.template, {
+        sourcepath: filename,
+        resourceDir: resDir,
+        locale: lo
+    });
+
+    // the file under en/US directory, it has to be located in the resource root
+    path = path.replace(/en\/([^A-Z])/, "$1");
```

Comment:
Should use the source locale rather than en/*

Reply:
It is for filtering the case like : `resources/en/strings.json` to `resources/strings.json` 
so  I need to exclude  like below
`resources/en/AU/strings.json`,  `resources/en/GB/strings.json` case

Reply:
If the source locale is always English for webos, this is okay.

Reply:
The source locale for webOS is always en-KR. so I think it's gonna be okay.